### PR TITLE
Porting TDistribution and NormalDistribution

### PR DESCRIPTION
**Pull request by dantleech**
As discussed in #32 this is an experiment in porting some of the Distribution functionality from the Java Apache Commons Math package.

### CODE REVIEW

File: Distribution/AbstractRealDistribution.php [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+     */
+    private $solverAbsoluteAccuracy = SOLVER_DEFAULT_ABSOLUTE_ACCURACY;
+
+    /**
+     * @param rng Random number generator.
+     * @since 3.1
+     */
+    public function __construct(Random $random)
+    {
+        $this->random = $random ? : new Random();
```
Comment (dantleech) [inline Distribution/AbstractRealDistribution.php]: We can use the native Hoa `Random` here.

**Comment by dantleech** [comment]
Very much a WIP.

**Comment by Hywan** [comment]
@dantleech Do not forget tests. Please, ping me when you need a review or if have any questions!

**Comment by dantleech** [comment]
Closing this. It basically became a case of needing to port not just one but multiple classes.

In PHPBench I ended up porting some code from the Python [scipy](https://github.com/scipy/scypy) and [numpy](https://github.com/numpy/numpy) math libraries, )i.e. [this KDE class](https://github.com/phpbench/phpbench/tree/master/lib/Math/Kde.php)).

I will try and improve and port my work from PHPBench to this repo in the future.

**Comment by Hywan** [comment]
Why not including `Hoa\Math` inside PHPBench?

**Comment by Hywan** [comment]
Or also [`Hoa\Bench`](http://central.hoa-project.net/Resource/Library/Bench)?

**Comment by dantleech** [comment]
I would rather include `Hoa\Math, but my port of that class is not as good as it could be (it only supports univariate data), I would want to put more effort into a PR here, and I will do so.

I don't think `hoa/bench` is applicable to PHPBench - PHPBench executes its measurements in a separate process with no external dependencies.

**Comment by Hywan** [comment]
PHPBench sounds far better than `hoa/bench`, I was mostly kidding :-). Good job by the way.

Ok to put more effort on `Hoa\Math`. I can help you. Tell me how and when. You can come on IRC if you want to discuss.

**Comment by dantleech** [comment]
ok, cool. I will try and port the existing maths functionality from PHPBench as soon as I can, initially using `numpy` as an example, and then we can take it from that PR or/and I will catch you on IRC :)

**Pull request closed without merging**